### PR TITLE
Minor manpage completion utility improvements

### PR DESCRIPTION
**Pull request by floam**
No longer truncates long lines with "` [See Man Page]`" suffix - reclaim 15 characters for actual  information. The ellipsis is enough - we are space constrained.

 `--verbose` output becomes a little less hard to follow:
 - diagnostics quoting input prints that input `repr()`'d with `%r` to be less
   confusing.
 - get rid of stupid `name()` class methods and use `type()`/`__class__.__name__`, the classes are already named such that this works nicely and it's communicating exactly what it is doing. Ensure usage of new-style (new as in post python 2.2) classes so this will not vary depending on if we run in Python 2 or 3.
 - spacing fixes

Convert left-quotes and right-quotes to that character in `deroff.py`, just like they are rendered in actual manpages. If this is uglier than how the manpages render anywhere (per existing comment) - I can't reproduce.

### CODE REVIEW

**Comment by faho** [comment]
LGTM, though in future I'd like this as two commits - one to do the visible change in normal behavior (which... why didn't I think of that???), another to do the cleanup.

File: share/tools/create_manpage_completions.py [hunk 1]
```diff
@@ -774,7 +774,8 @@ def parse_manpage_at_path(manpage_path, output_directory):
 
             # Output the magic word Autogenerated so we can tell if we can overwrite this
             built_command_output.insert(1, "# Autogenerated from man page " + manpage_path)
-            built_command_output.insert(2, "# using " + parser.__class__.__name__)
+            # built_command_output.insert(2, "# using " + parser.__class__.__name__) # XXX MISATTRIBUTES THE CULPABILE PARSER! Was really using Type2 but reporting TypeDeroffManParser
```
Comment (floam) [inline share/tools/create_manpage_completions.py]:
```
Considering /usr/share/man/man1/xcode-select.1
   Trying Type2ManParser
      Command is u'xcode-select'
   Trying Type3ManParser
      Command is u'xcode-select'
      Unable to find options section
   Trying TypeDarwinManParser
   Trying TypeDeroffManParser
   /usr/share/man/man1/xcode-select.1 parsed successfully
```
It was reporting the last parser that was used which ends up not being the same one that actually did the work. This was super annoying!

Comment (floam) [inline share/tools/create_manpage_completions.py]: Affected the following completions on my system FWIW:

```
🅸 ~/.l/s/f/generated_completions $ grep -r UNINDENT .
./clang++.fish:complete -c clang++ -s E --description 'Run the preprocessor stage. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o std --description 'Specify the language standard to compile for. UNINDENT NDENT 0.'
./clang++.fish:complete -c clang++ -o ansi --description 'Same as -std=c89. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o trigraphs --description 'Enable trigraphs. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o fms-extensions --description 'Enable support for Microsoft extensions. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o fborland-extensions --description 'Enable support for Borland extensions. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o fblocks --description 'Enable the "Blocks" language feature. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o arch --description 'Specify the architecture to build for. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o fvisibility --description 'This flag sets the default visibility level. UNINDENT NDENT 0.'
./clang++.fish:complete -c clang++ -l help --description 'Display available options. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o Xanalyzer --description 'Pass arg to the static analyzer. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o Xassembler --description 'Pass arg to the assembler. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o Xlinker --description 'Pass arg to the linker. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o Xpreprocessor --description 'Pass arg to the preprocessor. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -s o --description 'Write output to file. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o print-file-name --description 'Print the full library path of file. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o print-libgcc-file-name --description 'Print the library path for "libgcc. a". UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o print-prog-name --description 'Print the full program path of name. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o save-temps --description 'Save intermediate compilation results. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -o time --description 'Time individual commands. UNINDENT NDENT 0. 0.'
./clang++.fish:complete -c clang++ -s v --description 'Show commands to run and use verbose output. UNINDENT .'
./clang.fish:complete -c clang -s E --description 'Run the preprocessor stage. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o std --description 'Specify the language standard to compile for. UNINDENT NDENT 0.'
./clang.fish:complete -c clang -o ansi --description 'Same as -std=c89. UNINDEN NDENT 0. 0.'
./clang.fish:complete -c clang -o trigraphs --description 'Enable trigraphs. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o fms-extensions --description 'Enable support for Microsoft extensions. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o fborland-extensions --description 'Enable support for Borland extensions. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o fblocks --description 'Enable the "Blocks" language feature. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o arch --description 'Specify the architecture to build for. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o fvisibility --description 'This flag sets the default visibility level. UNINDENT NDENT 0.'
./clang.fish:complete -c clang -l help --description 'Display available options. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o Xanalyzer --description 'Pass arg to the static analyzer. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o Xassembler --description 'Pass arg to the assembler. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o Xlinker --description 'Pass arg to the linker. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o Xpreprocessor --description 'Pass arg to the preprocessor. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -s o --description 'Write output to file. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o print-file-name --description 'Print the full library path of file. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o print-libgcc-file-name --description 'Print the library path for "libgcc. a". UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o print-prog-name --description 'Print the full program path of name. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o save-temps --description 'Save intermediate compilation results. UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -o time --description 'Time individual commands.UNINDENT NDENT 0. 0.'
./clang.fish:complete -c clang -s v --description 'Show commands to run and use verbose output. UNINDENT .'
./xcode-select.fish:complete -c xcode-select -s h -l help --description 'Prints the usage message. UNINDENT NDENT 0. 0.'
./xcode-select.fish:complete -c xcode-select -s v -l version --description 'Prints xcode-select version information. UNINDENT NDENT 0. 0.'
./xcrun.fish:complete -c xcrun -s l -l log --description 'Print the full command line that is invoked. UNINDENT NDENT 0.'
./xcrun.fish:complete -c xcrun -l show-sdk-path --description 'Print the path to the selected SDK. UNINDENT NDENT 0. 0.'
./xcrun.fish:complete -c xcrun -l show-sdk-version --description 'Print the version number of the selected SDK. UNINDENT NDENT 0.'
```

**Comment by floam** [comment]
Actually, it's probably OK to rip out the thing that puts an ellipsis there entirely, right? Doesn't fish take care of truncating it wherever at runtime?

**Comment by krader1961** [comment]
I quickly skimmed the python changes and didn't see any problems so LGTM.
